docs/feature-debugging.md

```diff
@@ -0,0 +1,83 @@
+    * [Debugger for Chrome](https://marketplace.visualstudio.com/items?itemName=msjsdiag.debugger-for-chrome): Provides integrated client-side debugging in Code.
+
+## Launching and debugging a SAFE application in VS Code
+The SAFE template allows you to launch a SAFE Stack application with both client and server debugging enabled automatically.
+
+1. Create a new SAFE application using the [SAFE template](template-overview)
+1. Open VS Code in the folder containing your application
+1. Go to the Debug pane (`CTRL + Shift + D`)
+ 
+    <center><img src="../img/feature-debugging-1.png" style="height: 400px;"/></center>
+
+1. Ensure that **Debug SAFE App** is selected as the launch configuration
+1. Hit `F5` to launch the application, or hit the green play button in the launch configuration panel
+
+That's it. VS Code will now automatically:
+
+* Download all dependencies for both client and server
+* Build both the client and server
+* Start both client and server
+* Attach debuggers for both the client and server simultaneously!
+
+## The VS Code debugger
+This section explains the basics of debugging SAFE applications with VS Code. Refer to [here](https://code.visualstudio.com/docs/editor/debugging) to learn more about the VS Code debugger.
+
+### VS Code Tooling basics
+Once you start debugging a SAFE app, in the Terminal pane two Tasks will begin in parallel:
+
+* **Watch Client**: Builds and runs the client in hot module reload mode
+* **Build Server**: Builds and runs the server
+
+    <center><img src="../img/feature-debugging-2.png" style="height: 200px;"/></center>
+
+Low-level output from the command line of both client and server can be seen in the appropriate Terminal window. Observe that the floating Debug control contains two active debugging sessions:
+
+* **Debug Client**: Selects the Client debug session
+* **Debug Server**: Selects the Server debug session
+
+    <center><img src="../img/feature-debugging-3.png" style="height: 100px;"/></center>
+
+Selecting one or the other will instruct the Debug pane to show details for that "side" of the application e.g. breakpoints, call stack etc.
+
+### Setting breakpoints
+To set a breakpoint, simply highlight the line of code that you wish to stop on and hit `F9`, or click on the left-side gutter of the code window; a red dot will appear to show the breakpoint is set. You can see more on VS Code breakpoints [here](https://code.visualstudio.com/docs/editor/debugging#_breakpoints).
+
+Here you can see a breakpoint that has been hit in the `update` method of the sample application that is bundled with the SAFE template.
+
+<center><img src="../img/feature-debugging-4.png" style="height: 200px;"/></center>
+
+### Viewing debug output
+Debug output from Chrome will automatically be sent to the integrated Debug Console. This is especially useful for Elmish applications, as you can interrogate message and model updates directly in the IDE.
```

Comment:
I'm not a native speaker and had to lookup `interrogate`, maybe just something simpler: `watch` or `trace`